libs/database/src/lib/services/authentication.service.v2.ts

```diff
@@ -200,6 +200,17 @@ export class AuthenticationServiceV2 {
     }
   }
 
+  async getRefreshToken(refreshToken: string) {
+    const credential = await this.repository.findCredential({ refreshToken });
+    if (credential && credential.isRefreshTokenValid()) {
+      const account = await this.repository.findAccount({
+        _id: credential.account._id,
+      });
+      return this.login(credential, account);
+    }
+    throw CastcleException.INVALID_REFRESH_TOKEN;
+  }
+
```

Comment:
เขียนแบบ early return (guard clauses) + optional chaining ครับ

```suggestion
  async getRefreshToken(refreshToken: string) {
    const credential = await this.repository.findCredential({ refreshToken });
    if (!credential?.isRefreshTokenValid()) throw CastcleException.INVALID_REFRESH_TOKEN;
    const account = await this.repository.findAccount({
      _id: credential.account._id,
    });
    return this.login(credential, account);
  }
```